src/task_types/fuzzer_tasks.rs

```diff
@@ -0,0 +1,515 @@
+        DO_SETUP, GET_PROJECT_WITH_NO_MODULES, RUN_FUZZER, RUN_GENERATED_TESTS, SETUP_JSTESTFUZZ,
+    },
+    task_name::name_generated_task,
+};
+
+use super::{generated_suite::GeneratedSuite, multiversion::MultiversionService};
+
+/// Parameters for how a fuzzer task should be generated.
+#[derive(Default, Debug, Clone)]
+pub struct FuzzerGenTaskParams {
+    /// Name of task being generated.
+    pub task_name: String,
+    /// Name of build variant being generated on.
+    pub variant: String,
+    /// Resmoke suite for generated tests.
+    pub suite: String,
+    /// Number of javascript files fuzzer should generate.
+    pub num_files: String,
+    /// Number of sub-tasks fuzzer should generate.
+    pub num_tasks: u64,
+    /// Arguments to pass to resmoke invocation.
+    pub resmoke_args: String,
+    /// NPM command to perform fuzzer execution.
+    pub npm_command: String,
+    /// Arguments to pass to fuzzer invocation.
+    pub jstestfuzz_vars: Option<String>,
+    /// Should generated tests continue running after hitting error.
+    pub continue_on_failure: bool,
+    /// Maximum number of jobs resmoke should execute in parallel.
+    pub resmoke_jobs_max: u64,
+    /// Should tests be executed out of order.
+    pub should_shuffle: bool,
+    /// Timeout before test execution is considered hung.
+    pub timeout_secs: u64,
+    /// Requires downloading multiversion binaries.
+    pub require_multiversion_setup: Option<bool>,
+    /// Location of generated task configuration.
+    pub config_location: String,
+}
+
+impl FuzzerGenTaskParams {
+    /// Create parameters to send to fuzzer to generate appropriate fuzzer tests.
+    fn build_fuzzer_parameters(&self) -> HashMap<String, ParamValue> {
+        hashmap! {
+            "npm_command".to_string() => ParamValue::from(self.npm_command.as_str()),
+            "jstestfuzz_vars".to_string() => ParamValue::String(format!("--numGeneratedFiles {} {}", self.num_files, self.jstestfuzz_vars.clone().unwrap_or_default())),
+        }
+    }
+
+    /// Determine if these parameters are for a multiversion fuzzer.
+    fn is_multiversion(&self) -> bool {
+        self.require_multiversion_setup.unwrap_or(false)
+    }
+
+    /// Build the vars to send to tasks in the 'run tests' function.
+    ///
+    /// # Arguments
+    ///
+    /// * `generated_suite_name` - A generated suite to execute against.
+    /// * `version_combination` - Versions to start replica set with.
+    ///
+    /// # Returns
+    ///
+    /// Map of arguments to pass to 'run tests' function.
+    fn build_run_tests_vars(
+        &self,
+        generated_suite_name: Option<&str>,
+        version_combination: Option<&str>,
+    ) -> HashMap<String, ParamValue> {
+        let mut vars = hashmap! {
+            "continue_on_failure".to_string() => ParamValue::from(self.continue_on_failure),
+            "gen_task_config_location".to_string() => ParamValue::from(self.config_location.as_str()),
```

Comment:
Is this how enums are used in rust? If these are just strings, would you mind making them enums or constants?

Reply:
I switched these to constants. I'm not sure I understand your question about enums.

Reply:
Sounds good, we clarified on enums over Slack